content/blog/dark-mode-with-tailwind/index.md

```diff
@@ -0,0 +1,81 @@
+
+**TL;DR** I implemented dark mode *support* on my site using Tailwind with a one line addition 🌗🚀
+
+The eager eyed amongst you may have had a look under the hood of my website and seen what tech it’s built up on, you don’t need to look far to see the utility first approach I have taken to my CSS. Before you guys run wild and critisce the approach, let me in this post run through the reason why i chose it and why I love it.
+
+### Why Tailwind
+
+Tailwind CSS is a utility-first CSS framework for rapidly building custom solutions, it is relatively new and well-received in the dev community. To those who may not be familiar with what is meant by `utility first`, pretty much all the classes you will need already exists. For example, there are width and height utilities available which have an incremental approach allowing you to use classes in your mark-up that look like the below;
+
+```html
+<div class="d-block bg-white h-10 w-5 p-5"></div>
+```
+
+So, in the above code block, I have applied several utilities to the div element, all applied as several separated *utility classes*. In a more traditional setting, you would need define your own classes with all those properties, give those meaningful names and then use them. You may not see a big benefit in having all the utility classes pre-defined but what it allows you to do is rapidly build and prototype interfaces.
+
+This utility first approach can lead to long and ugly class names in your mark up, but it allows you to develop really quick and efficiently without writing any custom CSS. I believe it’s one of those concepts that you have to try out for yourself to appreciate.
+
+In the Tailwind Docs, Adam Wathan, the creator of Tailwind, has specifically created a page on utility first and outlined all its benefits and criticisms, I would recommended doing some [further reading](https://tailwindcss.com/docs/utility-first)
+
+My whole site uses this approach to styling and because of it I have had to use very little custom CSS. Pretty much all of it is predefined for me to use by Tailwind. Below is a snippet of my hero component on my site
+
+```html
+<h2 class="heading text-nord-1 text-xl sm:text-3xl md:text-4xl lg:text-5xl xl:text-6xl">
+  Hello, I'm Rahman.
+</h2>
+<h3 class="heading text-nord-3 text-base sm:text-lg md:text-xl lg:text-2xl xl:text-3xl">
+  Full-Stack Software Developer.
+</h3>
+```
+
+As you can see I have quite a few classes in my mark up, it leads to a more expressive way of styling as you know exactly what each class does, giving them single responsbilities. 
```

Comment:
As you can see I have quite a few classes in my mark up which leads to a more expressive way of styling as you know exactly what each class does - each class has a single responsbility.